plugins/in_kafka/in_kafka.c

```diff
@@ -128,10 +129,19 @@ static int process_message(struct flb_log_event_encoder *log_encoder,
 
     if (ret == FLB_EVENT_ENCODER_SUCCESS) {
         if (rkm->payload) {
-            if (try_json(log_encoder, rkm)) {
-                ret = flb_log_event_encoder_append_body_string(log_encoder,
-                                                               rkm->payload,
-                                                               rkm->len);
+            switch (ctx->data_format) {
```

Comment:
This can be simplified to avoid repeating the code without adding the switch, please do so.